src/main/java/ArrayDataException.java

```diff
@@ -0,0 +1,9 @@
+/**
+ * Custom java {@link  IllegalArgumentException} implementation, see <a href="https://highload.today/isklyucheniya-v-java/#:~:text=(fruits)%3B%20%7D%20%7D-,throws">source references</a>}.
+ */
+public class ArrayDataException extends IllegalArgumentException{
+    public ArrayDataException(String message) {
+        super(message);
+        super.printStackTrace();
```

Comment:
Better don't do that. If somewhere will be needed stack trace - will call it directly